### DISCUSSION TITLE
Feature Request: 100% Self-hosted Search Plugin via Stork

### DISCUSSION THREAD

**Discussion by ThatGuySam** [withastro/starlight#1198] [Feature Requests]
### What version of `starlight` are you using?

0.14.0

### What is your idea?

Implement a plugin to enable Stork Search so that you can host your site + search without an extra provider. 

https://github.com/jameslittle230/stork

### Why is this feature necessary?

For sites that may have a lot of traffic but not many documents or just as alternative Algolia. 

### Do you have examples of this feature in other projects?

I've implemented this into Astro on my own project, Does It ARM. 

The search goes through about 3400 documents in real-time and runs 100% on Netlify with no external servers. 

You can see a demo here: 
https://doesitarm.com/

Here's how it works in the browser: 
https://github.com/ThatGuySam/doesitarm/blob/6c0d536dc128d726aa2d6d375fea804b9301071e/helpers/stork/browser.js#L279

And the extra build step to index the documents: 
https://github.com/ThatGuySam/doesitarm/blob/6c0d536dc128d726aa2d6d375fea804b9301071e/helpers/stork/toml.js#L126
https://github.com/ThatGuySam/doesitarm/blob/6c0d536dc128d726aa2d6d375fea804b9301071e/helpers/stork/executable.js#L35

I can't give a hard commitment to do a pull request due to time right now, but happy to answer any questions on the implementation I did. 

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Hi @ThatGuySam, Starlight already includes self-hosted search by default! It's using Pagefind instead of Stork.

Algolia is an opt-in alternative only for those who prefer it.

Does Stork have any specific advantages over Pagefind, or does this meet your requirements?

**Reply by ThatGuySam** [reply]
Ah, my bad. I had never heard of Pagefind and it looks to be more feature rich than Stork right now so I can't really think of a reason for a Stork extension other than to give more options.

**Reply by kevinzunigacuellar** [reply]
If you would prefer to make your own integration with Stork, we welcome community plugins 💜. There is a section for them in the official docs https://starlight.astro.build/showcase/#community-plugins

**Reply by delucis** [reply]
Ah, good to know! Yes, as Kevin mentioned, community plugins are always welcome, but sounds like we probably wouldn't develop this as an official solution given Pagefind already provides a great default.

Thanks for the discussion!
